# Video Game Sales

`Matheus Raz (mrol@cin.ufpe.br)`

`João Paulo Lins (jplo@cin.ufpe.br)`

## Pre-processamento (etapa 1)

In [136]:
from IPython.display import display

import numpy as np
import pandas as pd
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score

In [80]:
df = pd.read_csv('vgsales.csv')
df.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [81]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
Rank            16598 non-null int64
Name            16598 non-null object
Platform        16598 non-null object
Year            16327 non-null float64
Genre           16598 non-null object
Publisher       16540 non-null object
NA_Sales        16598 non-null float64
EU_Sales        16598 non-null float64
JP_Sales        16598 non-null float64
Other_Sales     16598 non-null float64
Global_Sales    16598 non-null float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [82]:
df

Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
1          2                                  Super Mario Bros.      NES   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
4          5                           Pokemon Red/Pokemon Blue       GB   
5          6                                             Tetris       GB   
6          7                              New Super Mario Bros.       DS   
7          8                                           Wii Play      Wii   
8          9                          New Super Mario Bros. Wii      Wii   
9         10                                          Duck Hunt      NES   
10        11                                         Nintendogs       DS   
11        12                                      Mario Kart DS       DS   
12        13                        Pokemon Gold/Pokemon Silver       GB   
13        14                                            Wii Fit      Wii   
14        15                                       Wii Fit Plus      Wii   
15        16                                 Kinect Adventures!     X360   
16        17                                 Grand Theft Auto V      PS3   
17        18                      Grand Theft Auto: San Andreas      PS2   
18        19                                  Super Mario World     SNES   
19        20       Brain Age: Train Your Brain in Minutes a Day       DS   
20        21                      Pokemon Diamond/Pokemon Pearl       DS   
21        22                                   Super Mario Land       GB   
22        23                                Super Mario Bros. 3      NES   
23        24                                 Grand Theft Auto V     X360   
24        25                        Grand Theft Auto: Vice City      PS2   
25        26                      Pokemon Ruby/Pokemon Sapphire      GBA   
26        27                        Pokemon Black/Pokemon White       DS   
27        28        Brain Age 2: More Training in Minutes a Day       DS   
28        29                             Gran Turismo 3: A-Spec      PS2   
29        30                     Call of Duty: Modern Warfare 3     X360   
...      ...                                                ...      ...   
16568  16571                                        XI Coliseum      PSP   
16569  16572                                 Resident Evil 4 HD     XOne   
16570  16573                      Farming 2017 - The Simulation      PS4   
16571  16574         Grisaia no Kajitsu: La Fruit de la Grisaia      PSP   
16572  16575                   Scarlett: Nichijou no Kyoukaisen      PS2   
16573  16576                                Mini Desktop Racing      Wii   
16574  16577  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...      Wii   
16575  16578                              Neo Angelique Special      PSP   
16576  16579                                  Rugby Challenge 3     XOne   
16577  16580                                          Damnation       PC   
16578  16581                      Outdoors Unleashed: Africa 3D      3DS   
16579  16582                                  PGA European Tour      N64   
16580  16583                                          Real Rode      PS2   
16581  16584                                          Fit & Fun      Wii   
16582  16585                                    Planet Monsters      GBA   
16583  16586                                     Carmageddon 64      N64   
16584  16587                                   Bust-A-Move 3000       GC   
16585  16588                                             Breach       PC   
16586  16589                     Secret Files 2: Puritas Cordis       DS   
16587  16590                           Mezase!! Tsuri Master DS       DS   
16588  16591                                   Mega Brai

In [83]:
minor = df['Year'].min()
major = df['Year'].max()
print("Menor ano da base: %d"%(minor))
display(df[df['Year'] == minor])
print("Maior ano da base: %d"%(major))
display(df[df['Year'] == major])

Menor ano da base: 1980


Rank             Name Platform    Year     Genre   Publisher  NA_Sales  \
258    259        Asteroids     2600  1980.0   Shooter       Atari      4.00   
544    545  Missile Command     2600  1980.0   Shooter       Atari      2.56   
1766  1768          Kaboom!     2600  1980.0      Misc  Activision      1.07   
1969  1971         Defender     2600  1980.0      Misc       Atari      0.99   
2669  2671           Boxing     2600  1980.0  Fighting  Activision      0.72   
4025  4027       Ice Hockey     2600  1980.0    Sports  Activision      0.46   
5366  5368          Freeway     2600  1980.0    Action  Activision      0.32   
6317  6319           Bridge     2600  1980.0      Misc  Activision      0.25   
6896  6898         Checkers     2600  1980.0      Misc       Atari      0.22   

      EU_Sales  JP_Sales  Other_Sales  Global_Sales  
258       0.26       0.0         0.05          4.31  
544       0.17       0.0         0.03          2.76  
1766      0.07       0.0         0.01          1.15  
1969      0.05       0.0         0.01          1.05  
2669      0.04       0.0         0.01          0.77  
4025      0.03       0.0         0.01          0.49  
5366      0.02       0.0         0.00          0.34  
6317      0.02       0.0         0.00          0.27  
6896      0.01       0.0         0.00          0.24

Maior ano da base: 2020


Rank                    Name Platform    Year       Genre Publisher  \
5957  5959  Imagine: Makeup Artist       DS  2020.0  Simulation   Ubisoft   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
5957      0.27       0.0       0.0         0.02          0.29

In [84]:
df.drop(labels=5957,inplace=True)
display(df[df['Year'] == major])

Empty DataFrame
Columns: [Rank, Name, Platform, Year, Genre, Publisher, NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales]
Index: []

O jogo `Imagine: Makeup Artist` possui vendas espalhadas pelo mundo todo mesmo antes de seu lançamento, isso nos levou a pensar que fosse um tipo de pre-venda, porém após algumas pesquisas constatamos que o jogo foi lançado em 2009 e não possuia nenhuma previsão de lançamento para 2020, por isso tudo indica que seja um dado ruidoso, nos levando a questionar se os outros valores referentes a ele estariam realmente corretos. Optamos então por deletar esse valor do nosso dataset.

In [85]:
def verify(x):
    try:
        x = int(x)
        print("Deu certo!!")
    except:
        print("Valor",x,"não é número!!")

df.columns

Index(['Rank', 'Name', 'Platform', 'Year', 'Genre', 'Publisher', 'NA_Sales',
       'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales'],
      dtype='object')

In [86]:
# df['Year'].apply(verify)
qtdAnosNan = len(df[np.isnan(df['Year'])])
display("Quantidade de valores \"NaN\" na coluna Year: %d"%(qtdAnosNan))
df.dropna(inplace=True)
display("Quantidade após drop desses valores \"NaN\": %d"%(len(df[np.isnan(df['Year'])])))
df['Year'] = df['Year'].apply(lambda x : int(x)) # Transformando valores da coluna ano para inteiro.
display(df.head())

'Quantidade de valores "NaN" na coluna Year: 271'

'Quantidade após drop desses valores "NaN": 0'

Rank                      Name Platform  Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

### Tratamento de valores "NaN"

Testamos se havia valores `NaN`em todas as colunas do dataframe e vimos que só a coluna referente ao ano de lançamento do jogo que havia esses valores ruidosos, por serem apenas 271 linhas que não constavam, optamos por dropar eles, já que era um número pequeno comparado ao total de dados do dataframe (16.500 linhas).

In [87]:
print("qtd possíveis valores para plataforma: {}".format(len(df['Platform'].value_counts())))
print("qtd possíveis valores para genero: {}".format(len(df['Genre'].value_counts())))
print("qtd possíveis valores para publisher: {}".format(len(df['Publisher'].value_counts())))

qtd possíveis valores para plataforma: 31
qtd possíveis valores para genero: 12
qtd possíveis valores para publisher: 576


## Normalização

Normalizamos os valores de vendas de acordo com o maior valor de cada coluna em específica `value/maxValue`.

In [88]:
max_NA_Sales = df['NA_Sales'].max()
print(max_NA_Sales)
max_EU_Sales = df['EU_Sales'].max()
print(max_EU_Sales)
max_JP_Sales = df['JP_Sales'].max()
print(max_JP_Sales)
max_Other_Sales = df['Other_Sales'].max()
print(max_Other_Sales)
max_Global_Sales = df['Global_Sales'].max()
print(max_Global_Sales)

41.49
29.02
10.22
10.57
82.74


In [89]:
for i in df.columns[6:]:
    if(i == 'NA_Sales'):
        df[i] = df[i].apply(lambda x : x/max_NA_Sales)
    elif(i == 'JP_Sales'):
        df[i] = df[i].apply(lambda x : x/max_JP_Sales)
    elif(i == 'EU_Sales'):
        df[i] = df[i].apply(lambda x : x/max_EU_Sales)
    elif(i == 'Other_Sales'):
        df[i] = df[i].apply(lambda x : x/max_Other_Sales)
    elif(i == 'Global_Sales'):
        df[i] = df[i].apply(lambda x : x/max_Global_Sales)
df

Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
1          2                                  Super Mario Bros.      NES   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
4          5                           Pokemon Red/Pokemon Blue       GB   
5          6                                             Tetris       GB   
6          7                              New Super Mario Bros.       DS   
7          8                                           Wii Play      Wii   
8          9                          New Super Mario Bros. Wii      Wii   
9         10                                          Duck Hunt      NES   
10        11                                         Nintendogs       DS   
11        12                                      Mario Kart DS       DS   
12        13                        Pokemon Gold/Pokemon Silver       GB   
13        14                                            Wii Fit      Wii   
14        15                                       Wii Fit Plus      Wii   
15        16                                 Kinect Adventures!     X360   
16        17                                 Grand Theft Auto V      PS3   
17        18                      Grand Theft Auto: San Andreas      PS2   
18        19                                  Super Mario World     SNES   
19        20       Brain Age: Train Your Brain in Minutes a Day       DS   
20        21                      Pokemon Diamond/Pokemon Pearl       DS   
21        22                                   Super Mario Land       GB   
22        23                                Super Mario Bros. 3      NES   
23        24                                 Grand Theft Auto V     X360   
24        25                        Grand Theft Auto: Vice City      PS2   
25        26                      Pokemon Ruby/Pokemon Sapphire      GBA   
26        27                        Pokemon Black/Pokemon White       DS   
27        28        Brain Age 2: More Training in Minutes a Day       DS   
28        29                             Gran Turismo 3: A-Spec      PS2   
29        30                     Call of Duty: Modern Warfare 3     X360   
...      ...                                                ...      ...   
16568  16571                                        XI Coliseum      PSP   
16569  16572                                 Resident Evil 4 HD     XOne   
16570  16573                      Farming 2017 - The Simulation      PS4   
16571  16574         Grisaia no Kajitsu: La Fruit de la Grisaia      PSP   
16572  16575                   Scarlett: Nichijou no Kyoukaisen      PS2   
16573  16576                                Mini Desktop Racing      Wii   
16574  16577  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...      Wii   
16575  16578                              Neo Angelique Special      PSP   
16576  16579                                  Rugby Challenge 3     XOne   
16577  16580                                          Damnation       PC   
16578  16581                      Outdoors Unleashed: Africa 3D      3DS   
16579  16582                                  PGA European Tour      N64   
16580  16583                                          Real Rode      PS2   
16581  16584                                          Fit & Fun      Wii   
16582  16585                                    Planet Monsters      GBA   
16583  16586                                     Carmageddon 64      N64   
16584  16587                                   Bust-A-Move 3000       GC   
16585  16588                                             Breach       PC   
16586  16589                     Secret Files 2: Puritas Cordis       DS   
16587  16590                           Mezase!! Tsuri Master DS       DS   
16588  16591                                   Mega Brai

In [90]:
# Platform
def changeCategoric(column):
    teste = df[column].value_counts()
    binarys = []
    binarys_reference = teste.index
    count = 0
    for i in teste.index:
        zero = '0'
        binary = []
        for j in range(len(teste)):
            binary.append(zero)
        binary[count] = '1'
        count+=1
        current = ''
        for z in binary:
            current+=z
        binarys.append(current)
    values = []
    for i in binarys:
        values.append(i.encode('ascii'))
    return values,binarys_reference

convert_platform = changeCategoric('Platform')
print("PLATAFORMA CONVERTIDA\n\n\n\n")
display(convert_platform[0])
display(convert_platform[1])
print("GENERO CONVERTIDO\n\n\n\n")
convert_genre = changeCategoric('Genre')
display(convert_genre[0])
display(convert_genre[1])

PLATAFORMA CONVERTIDA






[b'1000000000000000000000000000000',
 b'0100000000000000000000000000000',
 b'0010000000000000000000000000000',
 b'0001000000000000000000000000000',
 b'0000100000000000000000000000000',
 b'0000010000000000000000000000000',
 b'0000001000000000000000000000000',
 b'0000000100000000000000000000000',
 b'0000000010000000000000000000000',
 b'0000000001000000000000000000000',
 b'0000000000100000000000000000000',
 b'0000000000010000000000000000000',
 b'0000000000001000000000000000000',
 b'0000000000000100000000000000000',
 b'0000000000000010000000000000000',
 b'0000000000000001000000000000000',
 b'0000000000000000100000000000000',
 b'0000000000000000010000000000000',
 b'0000000000000000001000000000000',
 b'0000000000000000000100000000000',
 b'0000000000000000000010000000000',
 b'0000000000000000000001000000000',
 b'0000000000000000000000100000000',
 b'0000000000000000000000010000000',
 b'0000000000000000000000001000000',
 b'0000000000000000000000000100000',
 b'0000000000000000000000000010000',
 

Index(['DS', 'PS2', 'PS3', 'Wii', 'X360', 'PSP', 'PS', 'PC', 'XB', 'GBA', 'GC',
       '3DS', 'PSV', 'PS4', 'N64', 'SNES', 'XOne', 'SAT', 'WiiU', '2600',
       'NES', 'GB', 'DC', 'GEN', 'NG', 'SCD', 'WS', '3DO', 'TG16', 'GG',
       'PCFX'],
      dtype='object')

GENERO CONVERTIDO






[b'100000000000',
 b'010000000000',
 b'001000000000',
 b'000100000000',
 b'000010000000',
 b'000001000000',
 b'000000100000',
 b'000000010000',
 b'000000001000',
 b'000000000100',
 b'000000000010',
 b'000000000001']

Index(['Action', 'Sports', 'Misc', 'Role-Playing', 'Shooter', 'Adventure',
       'Racing', 'Platform', 'Simulation', 'Fighting', 'Strategy', 'Puzzle'],
      dtype='object')

In [91]:
def changePlatform(x):
    value = -1
    for i in range(len(convert_platform[1])):
        if(convert_platform[1][i] == x):
            value = i
    return convert_platform[0][value]

def changeGenre(x):
    value = -1
    for i in range(len(convert_genre[1])):
        if(convert_genre[1][i] == x):
            value = i
    return convert_genre[0][value]

In [92]:
df['Platform'] = df['Platform'].apply(changePlatform)
df['Genre'] = df['Genre'].apply(changeGenre)

In [93]:
df

Rank                                               Name  \
0          1                                         Wii Sports   
1          2                                  Super Mario Bros.   
2          3                                     Mario Kart Wii   
3          4                                  Wii Sports Resort   
4          5                           Pokemon Red/Pokemon Blue   
5          6                                             Tetris   
6          7                              New Super Mario Bros.   
7          8                                           Wii Play   
8          9                          New Super Mario Bros. Wii   
9         10                                          Duck Hunt   
10        11                                         Nintendogs   
11        12                                      Mario Kart DS   
12        13                        Pokemon Gold/Pokemon Silver   
13        14                                            Wii Fit   
14        15                                       Wii Fit Plus   
15        16                                 Kinect Adventures!   
16        17                                 Grand Theft Auto V   
17        18                      Grand Theft Auto: San Andreas   
18        19                                  Super Mario World   
19        20       Brain Age: Train Your Brain in Minutes a Day   
20        21                      Pokemon Diamond/Pokemon Pearl   
21        22                                   Super Mario Land   
22        23                                Super Mario Bros. 3   
23        24                                 Grand Theft Auto V   
24        25                        Grand Theft Auto: Vice City   
25        26                      Pokemon Ruby/Pokemon Sapphire   
26        27                        Pokemon Black/Pokemon White   
27        28        Brain Age 2: More Training in Minutes a Day   
28        29                             Gran Turismo 3: A-Spec   
29        30                     Call of Duty: Modern Warfare 3   
...      ...                                                ...   
16568  16571                                        XI Coliseum   
16569  16572                                 Resident Evil 4 HD   
16570  16573                      Farming 2017 - The Simulation   
16571  16574         Grisaia no Kajitsu: La Fruit de la Grisaia   
16572  16575                   Scarlett: Nichijou no Kyoukaisen   
16573  16576                                Mini Desktop Racing   
16574  16577  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...   
16575  16578                              Neo Angelique Special   
16576  16579                                  Rugby Challenge 3   
16577  16580                                          Damnation   
16578  16581                      Outdoors Unleashed: Africa 3D   
16579  16582                                  PGA European Tour   
16580  16583                                          Real Rode   
16581  16584                                          Fit & Fun   
16582  16585                                    Planet Monsters   
16583  16586                                     Carmageddon 64   
16584  16587                                   Bust-A-Move 3000   
16585  16588                                             Breach   
16586  16589                     Secret Files 2: Puritas Cordis   
16587  16590                           Mezase!! Tsuri Master DS   
16588  16591                                   Mega Brain Boost   
16589  16592  Chou Ezaru wa Akai Hana: Koi wa Tsuki ni Shiru...   
16590  16593  Eiyuu Densetsu: Sora no Kiseki Material Collec...   
16591  16594                                Myst IV: Revelation   
16592  16595                                           Plushees   
16593  16596                 Woody Woodpecker in Crazy Castle 5   
16594  16597                      Men in Black II: Alien Escape   
16595  16598   SCORE International Baja 1000: The Offic

In [94]:
print('maior: {}'.format(df['Year'].max()))
print('menor: {}'.format(df['Year'].min()))

maior: 2017
menor: 1980


In [95]:
# print(convert_genre[0])
# print(convert_genre[1])
df[(df['Publisher'] == 'Electronic Arts') & (df['Genre'] == convert_genre[0][1])]

Rank                                Name  \
77        78                             FIFA 16   
112      113                             FIFA 14   
121      122                             FIFA 12   
124      125                             FIFA 15   
199      200                      FIFA Soccer 11   
211      212                       Madden NFL 06   
219      220                             FIFA 15   
221      222                             FIFA 17   
238      239                     Madden NFL 2005   
240      241                       Madden NFL 07   
256      257                             FIFA 14   
269      270                      FIFA Soccer 06   
275      276                             FIFA 12   
279      280                     Madden NFL 2003   
283      284                      FIFA Soccer 07   
303      304                    EA Sports Active   
334      335                    FIFA Soccer 2005   
347      348                      FIFA Soccer 10   
366      367                      FIFA Soccer 11   
392      393                    FIFA Soccer 2003   
416      417                             FIFA 16   
418      419                       Madden NFL 16   
440      441                      FIFA Soccer 08   
449      450                             FIFA 15   
451      452                     Madden NFL 2002   
493      494                             FIFA 14   
506      507                       Madden NFL 13   
513      514                             FIFA 16   
516      517                       Madden NFL 10   
534      535                          NBA Street   
...      ...                                 ...   
12724  12726                         FIFA Street   
12738  12740                      FIFA Soccer 06   
12778  12780                         NBA Live 07   
12881  12883                    FIFA Soccer 2003   
12893  12895            UEFA Euro 2004: Portugal   
12938  12940                             FIFA 17   
12944  12946     UEFA Champions League 2006-2007   
13040  13042                    FIFA Soccer 2004   
13247  13249                          Rugby 2005   
13262  13264                            Rugby 06   
13265  13267                 2002 FIFA World Cup   
13274  13276                      FIFA Soccer 10   
13358  13360                     FIFA Manager 12   
13516  13518                       FIFA Street 2   
13611  13613  FIFA Soccer 2003 (weekly jp sales)   
13851  13853         FIFA World Cup Germany 2006   
13956  13958                            Rugby 08   
13993  13995                              NHL 16   
14251  14254        FIFA World Cup: Germany 2006   
14789  14792                              NHL 07   
14814  14817                       Madden NFL 07   
15205  15208                      FIFA Soccer 09   
15443  15446                              NHL 09   
15528  15531                            Rugby 06   
15796  15799                              NHL 16   
15802  15805                    Football Academy   
15837  15840                      FIFA Soccer 08   
16012  16015     UEFA Champions League 2006-2007   
16127  16130                       Madden NFL 13   
16162  16165           Tiger Woods PGA Tour 2005   

                                 Platform  Year            Genre  \
77     b'0000000000000100000000000000000'  2015  b'010000000000'   
112    b'0010000000000000000000000000000'  2013  b'010000000000'   
121    b'0010000000000000000000000000000'  2011  b'010000000000'   
124    b'0000000000000100000000000000000'  2014  b'010000000000'   
199    b'0010000000000000000000000000000'  2010  b'010000000000'   
211    b'0100000000000000000000000000000'  2005  b'010000000000'   
219    b'0010000000000000000000000000000'  2014  b'010000000000'   
221    b'0000000000000100000000000000000'  2016  b'010000000000'   
238    b'0100000000000000000000000000000'  2004  b'010000000000'   
240    b'0100000000000000000000000000000'  2006  b'010000000000'   
256    b'00001000000000000000000000

## Aprendizagem (etapa 2)

In [137]:
df.corr()

Rank      Year  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank          1.000000  0.178128 -0.400318 -0.379150 -0.269333    -0.332742   
Year          0.178128  1.000000 -0.091301  0.006151 -0.169379     0.041157   
NA_Sales     -0.400318 -0.091301  1.000000  0.768925  0.451284     0.634519   
EU_Sales     -0.379150  0.006151  0.768925  1.000000  0.436377     0.726256   
JP_Sales     -0.269333 -0.169379  0.451284  0.436377  1.000000     0.290558   
Other_Sales  -0.332742  0.041157  0.634519  0.726256  0.290558     1.000000   
Global_Sales -0.426983 -0.074637  0.941270  0.903264  0.612773     0.747964   

              Global_Sales  
Rank             -0.426983  
Year             -0.074637  
NA_Sales          0.941270  
EU_Sales          0.903264  
JP_Sales          0.612773  
Other_Sales       0.747964  
Global_Sales      1.000000

In [145]:
df_RNA = df[['NA_Sales','EU_Sales']].copy()
classes_RNA = df['Other_Sales'].copy()
display(df_RNA)
classes_RNA

NA_Sales  EU_Sales
0      1.000000  1.000000
1      0.700892  0.123363
2      0.382020  0.443832
3      0.379610  0.379394
4      0.271632  0.306340
5      0.559171  0.077877
6      0.274283  0.318057
7      0.338154  0.317023
8      0.351651  0.243280
9      0.649072  0.021709
10     0.218607  0.379049
11     0.236443  0.260855
12     0.216920  0.212957
13     0.215474  0.276706
14     0.219089  0.296003
15     0.360810  0.170227
16     0.168956  0.319435
17     0.227284  0.013784
18     0.308026  0.129221
19     0.114485  0.319090
20     0.154736  0.155755
21     0.261027  0.093384
22     0.229935  0.118539
23     0.232104  0.182977
24     0.202699  0.189180
25     0.146059  0.134390
26     0.134249  0.113025
27     0.082912  0.184700
28     0.165100  0.175396
29     0.217643  0.147484
...         ...       ...
16568  0.000000  0.000000
16569  0.000241  0.000000
16570  0.000000  0.000345
16571  0.000000  0.000000
16572  0.000000  0.000000
16573  0.000241  0.000000
16574  0.000000  0.000000
16575  0.000000  0.000000
16576  0.000000  0.000345
16577  0.000000  0.000345
16578  0.000241  0.000000
16579  0.000241  0.000000
16580  0.000000  0.000000
16581  0.000000  0.000345
16582  0.000241  0.000000
16583  0.000241  0.000000
16584  0.000241  0.000000
16585  0.000241  0.000000
16586  0.000000  0.000345
16587  0.000000  0.000000
16588  0.000241  0.000000
16589  0.000000  0.000000
16590  0.000000  0.000000
16591  0.000241  0.000000
16592  0.000241  0.000000
16593  0.000241  0.000000
16594  0.000241  0.000000
16595  0.000000  0.000000
16596  0.000000  0.000345
16597  0.000241  0.000000

[16290 rows x 2 columns]

0        0.800378
1        0.072848
2        0.313150
3        0.280038
4        0.094607
5        0.054872
6        0.274361
7        0.269631
8        0.213813
9        0.044465
10       0.260170
11       0.181646
12       0.067171
13       0.203406
14       0.169347
15       0.157994
16       0.391675
17       1.000000
18       0.052034
19       0.193945
20       0.129612
21       0.039735
22       0.043519
23       0.130558
24       0.168401
25       0.047304
26       0.077578
27       0.111637
28       0.109745
29       0.124882
           ...   
16568    0.000000
16569    0.000000
16570    0.000000
16571    0.000000
16572    0.000000
16573    0.000000
16574    0.000000
16575    0.000000
16576    0.000000
16577    0.000000
16578    0.000000
16579    0.000000
16580    0.000000
16581    0.000000
16582    0.000000
16583    0.000000
16584    0.000000
16585    0.000000
16586    0.000000
16587    0.000000
16588    0.000000
16589    0.000000
16590    0.000000
16591    0.000000
16592    0

In [146]:
x_treino, x_teste, y_treino, y_teste = train_test_split(df_RNA,classes_RNA,test_size = 0.3)

model = LinearRegression()
# 2. Use fit
model.fit(x_treino, y_treino)
# 3. Check the score
score_regressao = model.score(x_teste, y_teste)
print("Porcentagem de acerto da regressão linear: {}%".format(score_regressao*100))

Porcentagem de acerto da regressão linear: 77.2092165010196%


### Hipótese: 

#### É possível prever o número de vendas em outras regiões do mundo baseado nas vendas da América do Norte, Europa e Japão?

A correlação entre as vendas das regiões da América do Norte e Europa com as vendas de outros lugares do mundo são bem fortes, o que implica serem valores relevantes entre eles. Através da execução da Regressão Linear, observamos que a chance de acerto foi de 77% aproximadamente, confirmando a hipótese de que é possível prever o número de vendas de um jogo em outras regioes, a partir das vendas desse jogo na América do Norte e na Europa.